### PR TITLE
jnigen/gradle: Add new task `jnigen` which regenerates jnigen artifacts.

### PR DESCRIPTION
**Pull request by PokeMMO**
* One line command `./gradlew jnigen` will regenerate all projects.
* Each BuildTarget gets a new task named `jnigenBuild${os}${64}`
*  `./gradlew jnigenBuild` task will build all builds that do not require a mac host.

**TODO**:
* [x] Potentially make `requireMacOSToBuild` a generic `requireOSToBuild` or an even more generic condition.
* [x] Support providing `APP_STL` and other extra lines to `Application.mk` `Android.mk`
* [x] Add better jar tasks for Desktop/Android/IOS.
* [ ] Add clean task.

### CODE REVIEW

**Comment by PokeMMO** [comment]
It might actually be simpler to just specify all possible options eg `bin:build/classes/java/main:target/classes` which would be legacy, gradle, and maven. I think javah will simply pick the first it finds, but will have to verify.

**Comment by Tom-Ski** [comment]
Maybe a custom plugin via buildSrc might be nice?

**Comment by PokeMMO** [comment]
> Maybe a custom gradle task that can be configured via buildSrc might be nice?

Initial PoC buldSrc gradle task created. Should be moved to it's own `gdx-jnigen-gradle` project and published as a proper plugin.

The `buildSrc/build.gradle` and `buildSrc/settings.gradle` could probably be done better, I only spent as much time on them as required to get it to work.

@Tom-Ski any feedback would be appreciated.

File: buildSrc/.classpath [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Tom-Ski) [inline buildSrc/.classpath]: Does eclipse not have support for buildSrc via buildship?

**Comment by Tom-Ski** [comment]
Perhaps we should think abut having gdx-jnigen extracted from the repo also with the plugin.

**Comment by MobiDevelop** [comment]
I did some initial work to move gdx-jnigen out - https://github.com/MobiDevelop/gdx-jnigen, needs the maven publish piece but I could move this repo into the libgdx org if we want to start that process.

File: buildSrc/.classpath (hunk 1 above)
Comment (PokeMMO) [reply buildSrc/.classpath]: I don't typically use buildship in eclipse, so I'm not sure. I can always remove the eclipse artifacts.

**Comment by PokeMMO** [comment]
I'll start by making a PR to MobiDevelop's gdx-jnigen project and setup a proper gradle plugin subproject in it.

**Comment by PokeMMO** [comment]
Initial gradle plugin PR created https://github.com/MobiDevelop/gdx-jnigen/pull/1

This PR has been updated to no longer modify any gdx-jnigen project files, but simply omit them from other project's *gradle* dependencies. No modifications to maven or included eclipse project files have been done.

Nothing breaking changes have been made, so maven/traditional Java based Build classes can continue to be used.

**Comment by MrStahlfelge** [comment]
Is this PR still needed/up to date?

**Comment by PokeMMO** [comment]
Not up to date, but should be updated and still merged.

**Comment by PokeMMO** [comment]
This PR has been rebased and updated. 

Should I also be removing `extensions/gdx-jnigen` with this or do that separately?
Should I be removing legacy build scripts (And ignoring them from git?) now or later?

Note: This is incompatible with full travis builds because travis can't run gradle on macOS due to missing Android SDK. We need to disable macOS builds until we migrate to Github Actions.

File: build.gradle [hunk 2]
```diff
@@ -25,14 +24,19 @@ allprojects {
         repositories {
             google()
             jcenter()
+            mavenLocal()
             mavenCentral()
             maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
         }
+        dependencies {
+            classpath "com.badlogicgames.gdx:gdx-jnigen-gradle:2.0.0-SNAPSHOT"
```
Comment (Tom-Ski) [inline build.gradle:32]: Lets do a release before we move to this?

**Review by Tom-Ski — APPROVED** [review]
Looks great, I'd just say lets release jnigen before using it here in core. 

Regarding killing old build scripts, I say yes.

File: build.gradle (hunk 2 above)
Comment (PokeMMO) [reply build.gradle:32]: Probably a good idea, there's nothing I'm aware of that needs done before release. There's a few small tweaks/features which might be good, but they're not worth holding it back for.

**Comment by Tom-Ski** [comment]
Also @PokeMMO what has this been tested on? 

If you haven't been able to test on all platforms, makes sense to get a checklist here and get some community help in testing anything you don't have device ability to test.

**Comment by PokeMMO** [comment]
This should create exact matches of the output ant scripts, so it should be fully compatible out of the box. Except I broke something when moving to `javac -h` from `javah`. Only noticed because none of the other test projects have relative project dependencies on classpath.

Github Actions has tested everything else on all platforms, except maybe iOS. I'll finish testing what I can first and then ask for help to verify the rest.

**Comment by PokeMMO** [comment]
Issues fixed, was InnerClasses being broken with `javac -h` and some classes having `compileOnly` preventing classpath from being populated correctly.

Linux/Windows/Android all compile correctly, will have to test macOS on GHA unless someone else can.

**Comment by PokeMMO** [comment]
I got a little carried away making this PR delete as many lines as possible, and removed the old gdx-jnigen extension too. If you'd rather me split that into it's own PR after this one let me know.

**Comment by PokeMMO** [comment]
@Tom-Ski
One last question before merge, is it acceptable that the build process require `./gradle jnigen` before running per-existing ant/maven based build for now?

**Comment by Tom-Ski** [comment]
Yeah thats ok. Its a step in the right direction to consolidate imo.

**Comment by Tom-Ski** [comment]
As a note to everyone, this is going to be merged on the weekend, so if you have any reservations, now is the time!

**Pull request merged**